amethyst_rendy/src/camera.rs

```diff
@@ -170,9 +187,19 @@ impl Orthographic {
     }
 
     /// Returns a mutable reference to the inner matrix representation of this projection.
+    ///
+    /// * panics when the supplied matrix is not invertible. See `Orthographic` and `Perspective`
+    /// for inputs.
+    #[inline]
+    pub fn set_matrix(&mut self, matrix: Matrix4<f32>) {
+        self.matrix = matrix;
+        self.inverse_matrix = matrix.try_inverse().expect("Matrix not invertible")
+    }
+
+    /// Returns a reference to the inner matrix representation of this projection.
     #[inline]
-    pub fn as_matrix_mut(&mut self) -> &mut Matrix4<f32> {
-        &mut self.matrix
+    pub fn as_inverse_matrix(&self) -> &Matrix4<f32> {
```

Comment:
Even if you don't intend to achieve feature parity with nalgebra, I think cached inverse matrix should be kept as an implemention detail for now.

Reply:
The matrix inversion is needed to compute the frustum (and likely other places as well). If we want to make it private, those operations (like getting the frustum) needs to be moved into the camera struct as well. I'm not sure if this is a good idea.